### ISSUE TITLE
7bit encoded email with line break

### ISSUE THREAD

**Issue by htulipe** [DockYard/elixir-mail#151]
Hello. Maybe a dumb question but I can't see how the lib can successfully parse a 7bit encoded email that contain line breaks. By successfully I mean without losing line breaks. 

## Version
mail 0.2.3
Erlang/OTP 24 [erts-12.3.2.6] [source] [64-bit] [smp:5:5] [ds:5:5:10] [async-threads:1]
Elixir 1.14.0 (compiled with Erlang/OTP 24)

## Test Case
Using the parse_email function [defined in parser test](https://github.com/DockYard/elixir-mail/blob/b5d4ab85491ee25d5a430801862eeca77fa1cf38/test/mail/parsers/rfc_2822_test.exs#L657):

```
parse_email("""
    To: user@example.com
    From: me@example.com
    Subject: Test Email
    Content-Transfer-Encoding: 7bit

    This is the body!
    It has more than one line
    """)
```

## Steps to reproduce

Run the above code

## Expected Behavior

The returned body should have some sort of line breaks:
`This is the body!\nIt has more than one line`

## Actual Behavior

The returned body no longer have line breaks:
`This is the body!It has more than one line`

Reading at the code, I see that the lib [joins the body lines using `\r\n`](https://github.com/DockYard/elixir-mail/blob/b5d4ab85491ee25d5a430801862eeca77fa1cf38/lib/mail/parsers/rfc_2822.ex#L427-L428) but the SevenBit parser [called just after](https://github.com/DockYard/elixir-mail/blob/b5d4ab85491ee25d5a430801862eeca77fa1cf38/lib/mail/parsers/rfc_2822.ex#L421) [drops them](https://github.com/DockYard/elixir-mail/blob/b5d4ab85491ee25d5a430801862eeca77fa1cf38/lib/mail/encoders/seven_bit.ex#L49). Am I missing something ?

Joining the body lines with `\n` instead of `\r\n` seems to fix the issue.

Thanks in advance

PS: I saw the [previous issue](https://github.com/DockYard/elixir-mail/issues/138) on the matter but could not find an answer there so I allowed myself to repost a new issue.

**Comment by bcardarella** [comment]
Is @andrewtimberlake 's answer in #138 not sufficient?

**Comment by htulipe** [comment]
I agree with Andrew's RFC understanding but the direct conclusion is that we can't send multi-line emails with this encoding. That can't be possible, I must be missing something.

**Comment by bcardarella** [comment]
@htulipe so your issue is not with the parsing but the compilation from the data structure into an email?

**Comment by htulipe** [comment]
My goal is to read an EML file and transform it in some data structure that my frontend end can then display.

**Comment by SergeyMosin** [comment]
> Is @andrewtimberlake 's answer in #138 not sufficient? ( https://github.com/DockYard/elixir-mail/issues/138#issuecomment-1103530953 )

First of all, thank you for your work on this module. However, I have the following question...

What should be the expected parsed message `body` for the following code according to [RFC 2045 §2.7](https://datatracker.ietf.org/doc/html/rfc2045#section-2.7) ?

```elixir
IO.inspect(
      Mail.parse([
        "From: a@b.tld",
        "To: c@d.tld",
        "Subject: test",
        "Content-Transfer-Encoding: 7bit", # or 8bit
        "",
        "line1",
        "line2"
      ])
    )
```
**Option A**: `line1\r\nline2`
1. :heavy_check_mark:  Data that is all represented as relatively short lines with 998 octets or less between CRLF line separation sequences.
2. :heavy_check_mark: No octets with decimal values greater than 127 are allowed and neither are NULs (octets with decimal value 0).
3. :heavy_check_mark: CR (decimal value 13) and LF (decimal value 10) octets only occur as part of CRLF line separation sequences.

**Option B**: `line1line2`
1. :x:  Data that is all represented as relatively short lines with 998 octets or less ~~between CRLF line separation sequences~~.
2. :heavy_check_mark: No octets with decimal values greater than 127 are allowed and neither are NULs (octets with decimal value 0).
3. :x: ~~CR (decimal value 13) and LF (decimal value 10) octets only occur as part of CRLF line separation sequences~~.

I personally lean towards **Option A**, but the `Mail.parse` function currently outputs **Option B** which seems to diverge from the RFC in points **1** and **3** because the "_CRLF line separation sequence_" is missing.

**Comment by andrewtimberlake** [comment]
I subsequently found out that 7bit decoding was removing line breaks indiscriminately and should only be removing those used to wrap lines exceeding the maximum length of 1000 chars
I have merged in a fix #164

**Comment by SergeyMosin** [comment]
Thank you for the quick fix. I think the same problem effects the `8bit` encoding as well.
Example:
```elixir
IO.inspect(
  Mail.parse([
    "From: a@b.tld",
    "To: c@d.tld",
    "Subject: test",
    "Content-Type: text/plain; charset=UTF-8",
    "Content-Transfer-Encoding: 8bit",
    "",
    "lÃ¯ne1",
    "lÃ¯ne2"
  ])
)
```
outputs this:
```
%Mail.Message{
  headers: %{
    "content-transfer-encoding" => "8bit",
    "content-type" => ["text/plain", {"charset", "UTF-8"}],
    "from" => "a@b.tld",
    "subject" => "test",
    "to" => ["c@d.tld"]
  },
  body: "lÃ¯ne1lÃ¯ne2",
  parts: [],
  multipart: false
}
```
no `\r\n` in the body

**Comment by andrewtimberlake** [comment]
Thanks, great catch. Fixed in #166
